lib/ansible/modules/cloud/vmware/vmware_host_dns.py

```diff
@@ -0,0 +1,469 @@
+- Tested on vSphere 6.5
+requirements:
+- python >= 2.6
+- PyVmomi
+options:
+  type:
+    description:
+    - Type of IP assignment. Either C(dhcp) or C(static).
+    - A VMkernel adapter needs to be set to DHCP if C(type) is set to C(dhcp).
+    type: str
+    default: 'static'
+    choices: [ 'dhcp', 'static' ]
+    required: False
+  device:
+    description:
+    - The VMkernel network adapter to obtain DNS settings from.
+    - The parameter is only required in case of C(type) is set to C(dhcp).
+    type: str
+  host_name:
+    description:
+    - The hostname to be used for the ESXi host.
+    type: str
+    required: True
+  domain:
+    description:
+    - The domain name to be used for the the ESXi host.
+    type: str
+    required: True
+  dns_servers:
+    description:
+    - A list of DNS servers to be used.
+    - The order of the DNS servers is important as they are used consecutively in order.
+    type: list
+    required: True
+  search_domains:
+    description:
+    - A list of domains to be searched through by the resolver.
+    type: list
+    required: False
+  verbose:
+    description:
+    - Verbose output of the DNS server configuration change.
+    - Explains if an DNS server was added, removed, or if the DNS server sequence was changed.
+    type: bool
+    required: false
+    default: false
+  esxi_hostname:
+    description:
+    - Name of the host system to work with.
+    - This parameter is required if C(cluster_name) is not specified.
+    type: str
+  cluster_name:
+    description:
+    - Name of the cluster from which all host systems will be used.
+    - This parameter is required if C(esxi_hostname) is not specified.
+    type: str
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Configure DNS for an ESXi host
+  vmware_host_dns:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    host_name: esx01
+    domain: example.local
+    dns_servers:
+      - 192.168.1.10
+      - 192.168.1.11
+    search_domains:
+      - subdomain.example.local
+      - example.local
+  delegate_to: localhost
+
+- name: Configure DNS for all ESXi hosts of a cluster
+  vmware_host_dns:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    cluster_name: '{{ cluster_name }}'
+    domain: example.local
+    dns_servers:
+      - 192.168.1.10
+      - 192.168.1.11
+    search_domains:
+      - subdomain.example.local
+      - example.local
+  delegate_to: localhost
+
+- name: Configure DNS via DHCP for an ESXi host
+  vmware_host_dns:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    type: dhcp
+    device: vmk0
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+result:
```

Comment:
Can we rename this to something more logical and related (like dns_config_result or dns_config_info) ?

Reply:
I've changed this to `dns_config_result`, I think this makes more sense. After all, it's the result of the DNS configuration while `dns_config_info` sounds like some information that's been discovered.